src/wayland/seat/mod.rs

```diff
@@ -49,7 +49,8 @@
 //! This module further defines the `"cursor_image"` role, that is assigned to surfaces used by clients
 //! to change the cursor icon.
 
-mod keyboard;
+#[allow(missing_docs)]
+pub mod keyboard;
```

Comment:
Is this right? This exposes `KeymapFile` to the consumers of the API. Do they need it somewhere? Perhaps you meant `pub(crate) mod keyboard`?  Using `pub(crate)` also removes the need for all the  `#[allow(missing_docs)]` notations that this PR adds.

If this is indeed meant to be used by the public, we should document it.

Reply:
This whole module heirarchy is just a mess and needs to be reordered

Reply:
Good point, I would move `KeymapFile` struct to `utils` module, if it is used by two different protocols. (pub(crate) question still applies even if it is in utils)

Reply:
Ah pub(crate) is correct, I wrote this before I knew how that worked. I will fix

Reply:
fixed